joomla/joomla-cms
# "Attempt to assign property 'cleanURL' on bool" fix

Pull Request for Issue #43443.

### Summary of Changes
This change addresses an error which can occur if Smart Search results are viewed while it is still indexing. When indexing an item, the 'object' column is initially reset to an empty value as can be seen [here](https://github.com/joomla/joomla-cms/blob/5768496747c98495ef8d5ca54130c18d87139d6b/administrator/components/com_finder/src/Indexer/Indexer.php#L359). The 'object' column is then set again once indexing for that item is complete. If the user tries to view Smart Search results during indexing, and their results contain the item that the indexer is currently working on, they will encounter the "Attempt to assign property 'cleanURL' on bool" 500 error. This is because it is trying to deserialize the empty 'object' and then set the 'cleanURL' variable, but because it is an empty object, is it is deserialized to 'false'.

This is especially prominent with large finder indexes where there is larger window for the user to encounter a result with an empty 'object' value.

The proposed fix ensures the Search model won't return any results which are missing their 'object' value, and so in the instances where the user would normally get a 500 error, the problematic result will just be omitted instead until it has been indexed.

### Testing Instructions
1. Start the Smart Search indexer in the Joomla admin panel.
2. While the indexer is running, access a Smart Search results page that includes the result with the manually emptied 'object' value.
3. Observe that a 500 error occurs.

As the issue only exhibits for a small window during the indexing process, this can be tested reliably by run the Smart Search indexer and then manually set the 'object' column of one of the rows in the 'jos_finder_links' table to an empty value. Go to a Smart Search results page which includes the result with the empty 'object' value and you should see the 500 error. This essentially forces it into the state that it is seen temporarily during index.

### Actual result BEFORE applying this Pull Request
Attempting to browse Smart Search results while indexing can result in the error: "Attempt to assign property 'cleanURL' on bool."

### Expected result AFTER applying this Pull Request
Browsing Smart Search results while the finder is indexing should no longer have a chance to result in an error.

### Link to documentations
Please select:
